dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/ProgramControllerTest.java

```diff
@@ -253,4 +255,60 @@ void testCopyProgramWithInvalidUid()
         assertEquals( "Program with id invalid could not be found.", response.getMessage() );
         assertEquals( E1005, response.getErrorCode() );
     }
+
+    @Test
+    void testCopyProgramWithUserWithNoAuthorities()
+    {
+        User userWithNoAuthorities = switchToNewUser( "test1" );
+        Set<String> authorities = userWithNoAuthorities.getUserRoles().stream()
```

Comment:
There is `getAllAuthorities()` on `User`, but maybe your implementation is nicer than the one currently in there :D